crypto/fipsmodule/curve25519/curve25519.c

```diff
@@ -132,14 +133,21 @@ void ED25519_keypair(uint8_t out_public_key[ED25519_PUBLIC_KEY_LEN],
 
   ed25519_keypair_pct(out_public_key, out_private_key);
 
+  FIPS_service_indicator_unlock_state();
   FIPS_service_indicator_update_state();
 }
 
 int ED25519_sign(uint8_t out_sig[ED25519_SIGNATURE_LEN],
                  const uint8_t *message, size_t message_len,
                  const uint8_t private_key[ED25519_PRIVATE_KEY_LEN]) {
+  FIPS_service_indicator_lock_state();
   boringssl_ensure_eddsa_self_test();
```

Comment:
Would it just be easier for the self-test functions to just always do an lock/unlock (in the FIPS cases)? Would simplify things.

Reply:
Or optionally pull the indicator update up out of the self_test functions (which you did already here) which eliminates the issue. I can fix these in my other Ed25519 if you want since there is overlap here between what we pick?